haystack/preview/dataclasses/document.py

```diff
@@ -66,10 +92,34 @@ class Document:
     def __str__(self):
         return f"{self.__class__.__name__}('{self.content}')"
 
+    def __eq__(self, other):
```

Comment:
As discussed in our call, let's loop through al the attributes of a document to check that every bit is exactly the same (including content).

Users can then use document_a.id == document_b.id if they just want to check the ids (calculated based on id_hash_keys) or they can use document_a == document_b to check for exact equality.